### ISSUE TITLE
How to limit row limit for admin.

### ISSUE THREAD

**Issue by xiaoyu-tamu** [hasura/graphql-engine#3469]
Is there anyway to set Limit number of rows for the default admin row?

**Comment by rikinsk** [comment]
@xiaoyu-tamu It is currently not possible to limit the number of rows returned for the admin role. the admin role is designed to have full access to the system. 

Can you explain your use case for this requirement in some more detail.

**Comment by xiaoyu-tamu** [comment]
@rikinsk  Yes, Sometimes we execute query in GRAPHIQL without limit parameter accidentally (it happens quite often if people use `ctrl + enter` to execute query). The query takes long time and may cause browser to crash since it try to render 10 million rows.

**Comment by delirehberi** [comment]
Maybe we can add a frontend control flow. That control runs when a query wanted to run without the `limit` parameter. Just asks the user:  `are you sure run to this command?`

**Comment by akshay5995** [comment]
@lexi-lambda What's the decision on how to handle this ? Is it going to be just an alert in the browser like @delirehberi I'm asking as it's labelled under `c/server` and not under `c/console`. I'd like to pick it up either ways. Thanks

**Comment by Benjamin-Philip** [comment]
@Stefmore02 Can I have a stab on this? If
all goes well, I'll probably have a PR in 2 - 3 weeks

**Comment by sachsom95** [comment]
If no one is working on it could I get this assigned and if possible point me to a starting point

**Comment by palash018** [comment]
@sachsom95 you got this assigned or are you working on it?

**Comment by sachsom95** [comment]
> @sachsom95 you got this assigned or are you working on it?

yet to get assigned @palash018

**Comment by palash018** [comment]
> 

if you get this assigned would you like to work together

**Comment by Stefmore02** [comment]
Hey folks @sachsom95 @palash018 we don't assign issues/PRs so feel free to start working on it. Once your PR is submitted, we'll review it (first come, first served).

**Comment by AadiDev005** [comment]
Hey @michael-land,

This is a real pain point - accidentally running unlimited queries in GraphiQL and crashing browsers happens way too often.

Since this is marked as server-side, a couple of approaches come to mind:

1. **Server-side default limit**: Add a configurable max row limit for admin queries (e.g., `HASURA_GRAPHQL_ADMIN_DEFAULT_LIMIT=1000`)

2. **Console warning**: Show a confirmation dialog when queries don't include explicit limits

Given this is labeled `c/server`, I think option 1 would be the cleanest solution. The admin role could still override with explicit `limit` parameters when needed, but would have a safety net for accidental unlimited queries.

Happy to work on this if no one else is actively tackling it.
